Sample/Kitchen Sink/DemonstrationTools/UnitConverter.swift

```diff
@@ -0,0 +1,184 @@
+                    convertButton1
+                }
+            }
+            .padding([.top], themeManager.current.padding.grid2x)
+
+            HStack {
+                VStack {
+                    HeadingView(
+                        title: demonstrationVM.adjustableConversionTitle,
+                        subTitle: demonstrationVM.adjustableConversionSubTitle)
+                    HStack {
+                        lengthInput
+                        lengthInputUnitPicker
+                    }
+
+                    HStack {
+                        calculationResult
+                        lengthOutputUnitPicker
+                    }
+                }
+                VStack {
+                    Spacer()
+                    convertButton2
+                        .padding([.bottom], themeManager.current.padding.grid2x)
+                }
+            }
+            .padding([.top], themeManager.current.padding.grid2x)
+        }
+        .padding([.leading, .trailing], themeManager.current.padding.grid2x)
+        .background(theme.color.background)
+        .navigationBarTitle(demonstrationVM.naivgationBarTitle)
+    }
+
+    var calculationResult: some View {
+        InputField(text: $demonstrationVM.outputValue, placeholder: demonstrationVM.calculatedField)
+            .textFieldStyle(.advisory)
+            .padding([.bottom], themeManager.current.padding.grid2x)
+    }
+
+    var lengthInput: some View {
+        InputField(text: $demonstrationVM.inputValue,
+                   placeholder: demonstrationVM.lengthPalceholder,
+                   bottomLabelConfig: BottomLabelConfig(""),
+                   filter: .doubleOnly)
+        .textFieldStyle(.default)
+    }
+
+    var lengthInputUnitPicker: some View {
+        MenuField(selection: $demonstrationVM.boundSelectionInput,
+                  options: UnitConverterViewModel.LengthType.allCases,
+                  placeholder: "")
+        .menuFieldStyle(.default)
+        .padding([.bottom], themeManager.current.padding.grid2x)
+    }
+
+    var lengthOutputUnitPicker: some View {
+        MenuField(selection: $demonstrationVM.boundSelectionOutput,
+                  options: UnitConverterViewModel.LengthType.allCases,
+                  placeholder: "")
+        .menuFieldStyle(.default)
+        .padding([.bottom], themeManager.current.padding.grid2x)
+    }
+
+    var kgInputField: some View {
+        InputField(text: $demonstrationVM.kgInputString,
+                   placeholder: demonstrationVM.kgHint,
+                   bottomLabelConfig: BottomLabelConfig(demonstrationVM.bottomKgLabel),
+                   formatter: { typedString in
+            guard let decimalValue = Decimal(string: typedString) else { return typedString }
+            return demonstrationVM.toString2DP(value: decimalValue)
+        }, filter: .doubleOnly)
+        .textFieldStyle(demonstrationVM.emptyFields ? DefaultTextFieldStyle.caution : DefaultTextFieldStyle.default)
```

Comment:
Empty fields is never updated within the view model, it's possible this was intended to be called on the convert button? But it would need to be aware of the `weightValuesSwapped` as one is expected to be empty 👍 

Also the default style needs to be changed to `.advisory` when it's the resultant value (e.g. converting from kg to lbs should have the lbs field as `.advisory` to display as white text that is uneditable 👍 )

Reply:
Is that the `if` statement in `checkForEmptyFields()`? If so yeah I see what you mean 🤔  perfect for it being in the viewmodel but yeah the style is growing quite complex haha. There might be a nicer way to refactor this, but at the moment seems to be working so happy to approve it for now and maybe we can get our heads together to think about if there's a cleaner way later on 👍